kerbauth.go

```diff
@@ -0,0 +1,180 @@
+package mssql
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"gopkg.in/jcmturner/gokrb5.v7/client"
+	"gopkg.in/jcmturner/gokrb5.v7/config"
+	"gopkg.in/jcmturner/gokrb5.v7/credentials"
+	"gopkg.in/jcmturner/gokrb5.v7/keytab"
+	"gopkg.in/jcmturner/gokrb5.v7/messages"
+	"gopkg.in/jcmturner/gokrb5.v7/spnego"
+	"gopkg.in/jcmturner/gokrb5.v7/types"
+)
+
+type Krb5ClientState int
+
+const (
+	ContextFlagREADY = 128
+	/* initiator states */
+	InitiatorStart Krb5ClientState = iota
+	InitiatorRestart
+	InitiatorWaitForMutal
+	InitiatorReady
+)
+
+type krb5Auth struct {
+	username          string
+	realm             string
+	service           string
+	password          string
+	port              string
+	krb5ConfFile      string
+	krbFile           string
+	initkrbwithkeytab string
+	krb5Client        *client.Client
+	state             Krb5ClientState
+}
+
+var clientWithKeytab = client.NewClientWithKeytab
+var loadCCache = credentials.LoadCCache
+var clientFromCCache = client.NewClientFromCCache
+var spnegoNewNegToken = spnego.NewNegTokenInitKRB5
+var spnegoToken spnego.SPNEGOToken
+var spnegoUnmarshal = spnegoToken.Unmarshal
+var kt = &keytab.Keytab{}
+var ktUnmarshal = kt.Unmarshal
+
+var negTokenMarshal = func(negTok spnego.NegTokenInit) ([]byte, error) {
+	return negTok.Marshal()
+}
+var getServiceTicket = func(cl *client.Client, spn string) (messages.Ticket, types.EncryptionKey, error) {
+	return cl.GetServiceTicket(spn)
+}
+
+func getKRB5Auth(user, service, krb5Conf, krbFile, initkrbwithkeytab, password string) (auth, bool) {
+	if krb5Conf == "" {
+		krb5Conf = "/etc/krb5.conf"
```

Comment:
This must be behind some type of OS flag, either by file or by runtime.OS string switch. This makes no sense on window and unsure about OSX.